README.md

```diff
@@ -0,0 +1,152 @@
+
+
+## Example
+
+- User Inputted Keyword: Data Structure
+- Output:
+
+![Screenshot](https://github.com/Forward-UIUC-2021F/keyword-usage-within-domain/blob/fbfb0cd4160400a8d12e9ba8945ce718be663fb6/images_readme/Screen%20Shot%202021-08-26%20at%2011.08.00%20PM.png)
+
+![Screenshot](https://github.com/Forward-UIUC-2021F/keyword-usage-within-domain/blob/fbfb0cd4160400a8d12e9ba8945ce718be663fb6/images_readme/Screen%20Shot%202021-09-10%20at%204.25.32%20PM.png)
+
+- User Inputted Keyword: Machine Learning
+- Output:
+
+![Screenshot](https://github.com/Forward-UIUC-2021F/keyword-usage-within-domain/blob/fbfb0cd4160400a8d12e9ba8945ce718be663fb6/images_readme/Screen%20Shot%202021-08-26%20at%2011.51.09%20PM.png)
+
+
+
+
+## Methodology:
+
+### Whoosh
+ 
+Whoosh is a fast, featureful full-text indexing and searching library implemented in pure Python. Programmers can use it to easily add search functionality to their applications and websites. Every part of how Whoosh works can be extended or replaced to meet your needs exactly.
+
+We can use Whoosh in order to access certain bits of information that we need. The input is the Documents from the Arxiv Dataset.
+
+
+### Extracting Information
+
+We are trying to generate a timeline of a specific keyword. Using the Arxiv Dataset of CS Google Scholar articles, we can gather the abstract, the keyword, the year the document was uploaded, as well as the categories. After gathering this information, we can add it into a MySQL database.
+
+In order to associate a keyword with each Scholar article we will use the abstract of each article. By iterating through the Scholar Dataset, we will determine what keywords appear in the abstract. From there, we will increment the keyword-year frequency. 
+
+In order to generate other information, we will also note the categories that are associated with that specific keyword. Through doing this, we can determine the history of the usage of that keyword within that category.
+
+### Database 
+
+We can create a database in order to generate a timeline of a specific keyword and other needed information. We can create a table with attributes of the keyword itself, the year, the category code, and it’s frequency. 
+
+For both of these operations, we will use MySQL as the database program. 
+
+### Table With Fields
+
+Keyword, Year, Category Code, Frequency
+
+### Display
+
+With the table, we can display them in different ways. We can use SQL’s group by function to display the two different outputs. With the keyword, year, and frequency, we can display it in a timeline, similar to the Ngram Viewer. With the categories associated with the keyword, we can display it as a sort of bar graph.
+
+
+
+
+
+
+# Functional Design
+
+- user can pass in a keyword within CS domain within a given year and find frequency. if no frequency for that year, return closest frequency
```

Comment:
Can you combine the functions `get_frequency_of_year` and `get_frequency_of_category` into one? You can make `year` and `category` optional arguments. 

I would also suggest renaming the combined function so that it is clear we are returning the frequency of the keyword (maybe something like `get_keyword_frequency(keyword, year=None, category_code=None)`